src/main/java/org/sunbird/profile/service/ProfileServiceImpl.java

```diff
@@ -738,4 +735,148 @@ private String assignUserRole(Map<String, Object> requestBody, List<String> exis
 		}
 		return errMsg;
 	}
+	@Override
+	public SBApiResponse userSignup(Map<String, Object> request) throws Exception {
+		boolean retValue = false;
+		SBApiResponse response = createDefaultResponse(Constants.API_PROFILE_SIGNUP);
+		String errMsg = validateSignupRequest(request);
+		if (!StringUtils.isEmpty(errMsg)) {
+			response.getParams().setErrmsg(errMsg);
+			response.setResponseCode(HttpStatus.BAD_REQUEST);
+			return response;
+		}
+		try {
+			Map<String, Object> requestData = (Map<String, Object>) request.get(Constants.REQUEST);
+			Map<String, Object> reqObj = new HashMap<>();
+			Map<String, Object> requestBody = new HashMap<String, Object>();
+			requestBody.put(Constants.EMAIL, requestData.get(Constants.EMAIL));
+			requestBody.put(Constants.FIRSTNAME, requestData.get(Constants.FIRST_NAME));
+			requestBody.put(Constants.LASTNAME, requestData.get(Constants.LAST_NAME));
+			requestBody.put(Constants.EMAIL_VERIFIED, true);
+			reqObj.put(Constants.REQUEST, requestBody);
+				Map<String, Object> readData = (Map<String, Object>) outboundRequestHandlerService.fetchResultUsingPost(
+						props.getSbUrl() + props.getLmsUserSignUpPath(), reqObj, getDefaultHeaders());
+				if (Constants.OK.equalsIgnoreCase((String) readData.get(Constants.RESPONSE_CODE))) {
+					Map<String, Object> result = (Map<String, Object>) readData.get(Constants.RESULT);
+					String userId = (String) result.get(Constants.USER_ID);
+					request.put(Constants.USER_ID, userId);
+					Map<String, Object> userData = getUsersReadData(userId, StringUtils.EMPTY, StringUtils.EMPTY);
+					if (CollectionUtils.isNotEmpty(Collections.singleton(userData))) {
```

Comment:
check if the map is not empty, instead of coverting it to list and then checking if list is empty

Reply:
resolved